### PR TITLE
Add support to disable b-frames for h265 video codec

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add support to disable b-frames on H265 video codec

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

**Review by patrick-tolosa — APPROVED** [review]

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt [hunk 1]
```diff
@@ -283,6 +291,41 @@ class H264Codec private constructor(private val profile: VideoCodecProfile? = nu
...
+            this.bFrames = bFrames
+        }
+
+        private fun setAutoProfileAndLevel(setAutoProfleAndLevel: Boolean) {
+            if (setAutoProfleAndLevel) {
+                this.profile = null;
+                this.level = null
+            } else {
+                this.profile = VideoCodecProfile.auto()
+                this.level = "auto"
```
Comment (const-cloudinary) [inline url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:320]: Why not `VideoCodecLevel.auto()`?

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt [hunk 2]
```diff
@@ -283,6 +291,41 @@ class H264Codec private constructor(private val profile: VideoCodecProfile? = nu
...
+        private var profile: VideoCodecProfile? = null
+        private var level: Any? = null
+        private var bFrames: Boolean = true
+
+        fun bFrames(bFrames: Boolean) = apply {
+            setAutoProfileAndLevel(bFrames)
+            this.bFrames = bFrames
+        }
+
+        private fun setAutoProfileAndLevel(setAutoProfleAndLevel: Boolean) {
```
Comment (const-cloudinary) [inline url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:314]:
```suggestion
        private fun setAutoProfileAndLevel(setAutoProfileAndLevel: Boolean) {
```

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt [hunk 3]
```diff
@@ -283,6 +291,41 @@ class H264Codec private constructor(private val profile: VideoCodecProfile? = nu
...
+            if (setAutoProfleAndLevel) {
```
Comment (const-cloudinary) [inline url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:315]: The logic here is a bit confusing, when passing `setAutoProfileAndLevel` = true, I would expect it to have the `auto` values, and not `null`

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt (hunk 1 above)
Comment (adimiz1) [reply url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:320]: Unlike profile which is type `VideoCodecProfile` level is `Any` and that's why I gave it a `String` value

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt (hunk 3 above)
Comment (adimiz1) [reply url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:315]: Will fix

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt (hunk 2 above)
Comment (adimiz1) [reply url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:314]: Fixing, thanks

File: url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt (hunk 1 above)
Comment (const-cloudinary) [reply url-gen/src/main/kotlin/com/cloudinary/transformation/transcode/TranscodeActions.kt:317]: I see no reason not to add it to the enum

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**